main.py

```diff
@@ -109,23 +115,21 @@ def load_chunks(inventory, lg, ck):
   return chks
 
 
-def load_vectors(docs):
+def load_vector(d):
   embed = OpenAIEmbeddings()
-  msg = f"Loading vectors..."
-  sz = len(docs)
-  prog = st.progress(0.0, text=msg)
-  vectors = []
-  for i, d in enumerate(docs, start=1):
-    vectors.append(embed.embed_query(d.page_content))
-    prog.progress(i/sz, text=msg)
-  prog.empty()
-  return vectors
+  result = embed.embed_query(d.page_content)
+  return result
 
 
-@st.cache_resource(show_spinner=False)
+@st.cache_resource(show_spinner="Loading Vectors...")
 def select_docs(dt, ch, lg, lm, ck, ct):
   docs = load_chunks(inventory, lg, ck)
-  vectors = load_vectors(docs)
+  docs_list = [(d,) for d in docs]
+
+  with ThreadPool(THREAD_COUNT) as pool:
+    vectors = pool.starmap(load_vector, docs_list)
+  st.success('Vectors loaded!')
```

Comment:
We don't need this, because it will be persistent on the page, if not in cache. When the spinner stops, the summary will start to load, so no need for this info.

Reply:
Is this comment about the threadpool or the 'Vectors loaded!' message?

Reply:
This was about the `st.success()` message.